kaos/algorithm/visibility_finder.py

```diff
@@ -335,7 +346,54 @@ def determine_visibility(self, error=0.1, tolerance_ratio=0.1, max_iter=100):
         # If the loop terminates and an access end was still not found that means that point should
         # still be visible at the end of the period.
         # NOTE: subinterval_end would also work here but is difficult to test.
-        if access_start is not None:
+        if (access_start is not None) and (access_start < end_time):
+            if self.visibility(end_time) <= 0:
+                raise VisibilityFinderError("Visibility interval started at {} "
+                                            "but did not end at {}".format(access_start, end_time))
+            sat_accesses.append((access_start, end_time))
+
+        # TODO: switch this to log
+        print("average h: {}".format((end_time-start_time)/interval_num))
+
+        return sat_accesses
+
+
+    def determine_visibility_brute_force(self, step=5):
```

Comment:
Can't blame me. It's been up for review for 10 days now. On branch for 2-3 weeks!

Reply:
I think this method should be placed in test not kaos.

Reply:
I disagree. This is not a test function; It can do exactly what determine_visibility() does with arbitrary accuracy. The user might want to switch to using this in some cases, I know we have...

Reply:
Whats the opinin on this @KMC-70/kaos-overlords ?

Reply:
This function is fine to be here, however, we should explicitly state it should not be used unless we plan on benching